roles/update_grub/README.md

```diff
@@ -0,0 +1,34 @@
+update_grub
+=================
+
+Description
+-----------
+The `update_grub` Ansible role is designed to manage the GRUB (Grand Unified Bootloader) configuration and configure HugePages settings on Ubuntu servers. It allows you to apply specific parameters to improve system performance and efficiently manage HugePages.
+
+Requirements
+------------
+Before using this role, ensure the following requirements are met:
+
+- The role assumes that the target servers are running Ubuntu.
+- Make sure you have SSH access to the target servers.
+
+Role Variables
+--------------
+The `update_grub` role utilizes the following variables defined in custom host group `group_vars/k8s-vms/grub.yml` to update the grub config on specific group of servers depending on host usage:
+
+```
+grub_config:
+  cmdline_linux_default:
+    - 'iommu=pt'
+    - 'intel_iommu=on'
+    - 'default_hugepagesz=1G'
+    - 'transparent_hugepage=never'
+    - 'hugepages=20'
+    - 'hugepagesz=1G'
+
+nr_hugepages: 25
+```
+
+The `grub_config` variable allows you to specify various parameters to be added to the `GRUB_CMDLINE_LINUX_DEFAULT` option in the `/etc/default/grub` file. These parameters are designed to optimize system performance and hardware support.
+
+The `nr_hugepages` variable sets the number of HugePages that will be configured in the `/etc/sysctl.conf` file. HugePages provide significant benefits for certain applications, such as those running on virtual machines or memory-intensive workloads.
```

Comment:
architectonic note:
We have here update_grub and sysctl_config both configuring kernel parametrs at runtime. Would it be possible to move nr_hugepages: 25 to sysctl_config configuration?

Reply:
Discussed, I finally agree to keep it here, but let's change configuration following way
```yaml
update_grub:
  grub_config:
    cmdline_linux_default:
      - 'iommu=pt'
      - 'intel_iommu=on'
      - 'default_hugepagesz=1G'
      - 'transparent_hugepage=never'
      - 'hugepages=512'
      - 'hugepagesz=1G'
  nr_hugepages: 512
```
so nr_hugepages and grub_config is under key `update_grub` (ansible module name)

Reply:
Yesterday I discussed about update_grub role with @berkas1 and we concluded that it is quite a dup of already existing role `grub`.
In order to merge majority of this bigger PR, I propose to strip `update_grub` completely from this PR and we will discuss in what form it will be re-added in separate PR.

Reply:
To be more concrete:
Already existing `grub` role is used to conditionally switch kernel's cgroups to v1 (which we do for compute nodes)
Proposed `update_grub` does completely same thing (kernel configuration via grub) but different way.

It would be great to merge your functionality into `grub` role the way we introduce list of additional kernel parameters to be added.

As a side-effect it means huge pages sysctl configuration moves to `sysctl_config` role configuration.

As this is more work I propose to do it in separate PR.

Reply:
Alright I will completely remove `update_grub` role and update this PR . 
Yes I do agree we can merge `grub` and `update_grub` together. 
We decided to not to use sysctl config anymore to configure hugepages. Grub config should enough for it.
```
---
grub_config:
  cmdline_linux_default:
    - 'iommu=pt'
    - 'intel_iommu=on'
    - 'default_hugepagesz=2M'
    - 'transparent_hugepage=never'
    - 'hugepages=512'
    - 'hugepagesz=2M'
```